packages/@ember/-internals/routing/lib/services/router.ts

```diff
@@ -309,6 +310,19 @@ export default class RouterService extends Service {
     let { routeName, models, queryParams } = extractRouteArgs(args);
     let routerMicrolib = this._router._routerMicrolib;
 
+    // When using isActive() in a getter, we want to entagle with the auto-tracking system
+    // for example,
+    // in
+    // get isBarActive() {
+    //   return isActive('foo.bar');
+    // }
+    //
+    // you'd expect isBarActive to be dirtied when the route changes.
+    //
+    // https://github.com/emberjs/ember.js/issues/19004
+    consumeTag(tagFor(this, 'currentURL'));
+    consumeTag(tagFor(this, 'currentRouteName'));
```

Comment:
These are both aliased computed properties, I'd prefer to entangle the actual root state itself (not the CP's here).

Thoughts @pzuraq?

Reply:
I think that's true, but I'm unsure if it's practical given the state of the router right now, unless you mean entangling `_router.currentURL`. I think entangling the `currentURL` state makes sense in the meantime, until we refactor the router. I'm not sure why we would have to entangle `currentRouteName` as well though, we should try to our dependencies here I think.

Reply:
> unless you mean entangling _router.currentURL

Ya, I mostly mean: don't use the alias CP, entangle the root state from `router:main` directly.

Reply:
done & pushed